src/css_parser.rs

```diff
@@ -1091,10 +1122,22 @@ pub enum CssBackgroundParseError<'a> {
     ImageParseError(CssImageParseError<'a>),
 }
 
-impl_from!(CssDirectionParseError, CssBackgroundParseError::DirectionParseError);
-impl_from!(CssGradientStopParseError, CssBackgroundParseError::GradientParseError);
-impl_from!(CssShapeParseError, CssBackgroundParseError::ShapeParseError);
-impl_from!(CssImageParseError, CssBackgroundParseError::ImageParseError);
+impl_display!{ CssBackgroundParseError<'a>, {
+    Error(e) => e,
+    InvalidBackground(val) => format!("Invalid value: \"{}\"", val),
+    UnclosedGradient(val) => format!("Unclosed gradient: \"{}\"", val),
+    NoDirection(val) => format!("No direction: \"{}\"", val),
+    TooFewGradientStops(val) => format!("Too few gradient-stops: \"{}\"", val),
+    DirectionParseError(e) => format!("Direction parse error: {}", e),
```

Comment:
Better: "Could not parse gradient direction: {}"

"Direction error" is not a very descriptive error.